### PR TITLE
Add GitHub Actions

### PR DESCRIPTION
**Pull request by samellem**
Add GitHub action (.github/workflows/ci.yml) to build and test AutoDP code. To facilitate this, add Bazel BUILD files and WORKSPACE definition. Also add configuration for running pytype checks over the repository and fill in a missing case in fdp_fdp_grad_to_approxdp from converter.py to allow those checks to pass.

### CODE REVIEW

**Comment by samellem** [comment]
See example successful run at https://github.com/samellem/autodp/actions/runs/685447186.

**Review by yuxiangw — COMMENTED** [review]
What is the reason behind not allowing Python 3.9 ?  Or is this only about testing python 3.6, 3.7, 3.8?

File: setup.py [hunk 1]
```diff
@@ -14,10 +25,10 @@
    author='Yu-Xiang Wang',
    author_email='yuxiangw@cs.ucsb.edu',
    url='https://github.com/yuxiangw/autodp',
-   download_url = 'https://github.com/yuxiangw/autodp/archive/v0.1.1.tar.gz',   
+   download_url = 'https://github.com/yuxiangw/autodp/archive/v0.1.1.tar.gz',
```
Comment (yuxiangw) [inline setup.py:28]: This has changed to v0.2. I will see if I can directly edit  setup.py in your branch.

**Comment by samellem** [comment]
It's just explicitly testing 3.6 through 3.8, not disallowing 3.9. We could probably add 3.9 as well if you'd like.

**Comment by yuxiangw** [comment]
> It's just explicitly testing 3.6 through 3.8, not disallowing 3.9. We could probably add 3.9 as well if you'd like.

Thanks.  It is great that we are setting up Bazel. I think it is not necessary to add 3.9 for now  (running over all different versions may make the CI somewhat slow).

**Pull request merged**